### ISSUE TITLE
golang indent issues when auto-pairs = false

### ISSUE THREAD

**Issue by salarkhan** [helix-editor/helix#7910]
### Summary

when writing expressions, i expect an outdent when i input the matching pair. in some cases, the closing brace is in line with the content of the expression:
<img width="164" alt="Screenshot 2023-08-11 at 8 35 08 AM" src="https://github.com/helix-editor/helix/assets/3118416/454fbd13-f2c4-40ca-acd5-06268629c2c4">

in other cases like creating a function, i expect an indent after opening the brace and hitting enter, yet it does not.  

all of these scenarios are resolved if i close the braces immediately, and _then_ fill in the content. 

### Reproduction Steps

https://github.com/helix-editor/helix/assets/3118416/951eeddb-8826-4c03-ade0-3cdd182f1e1d

### Platform

macOS

### Terminal Emulator

kitty 0.28.0

### Helix Version

helix 23.05

**Comment by woojiq** [comment]
Iirc that's how a tree-sitter works. This is not specific to golang. If you select your function without the closing brace and run `:tree-sitter-subtree` you will see that there is an error when parsing the source code.

**Comment by woojiq** [comment]
Probably duplicate of https://github.com/helix-editor/helix/issues/6044

**Comment by pascalkuthe** [comment]
Yeah the closing bracket not causing a decent without autopairs  is both known and expected and would be covered by #6044.

We do sometimes work around situations like this by also matching error nodes in the tree sitter indent captures.

CC @Triton171

**Comment by Triton171** [comment]
I think at least for simple scenarios it should be possible to fix this my matching error nodes. A possible pattern would be:
```scm
(ERROR
    "{" @indent @extend
)
```
This correctly indents a function even if the closing `}` is missing. However, an incomplete if statement inside the function is not indented correctly. Finding queries that work correctly for multiple incomplete nesting levels seems non-trivial since the tree-sitter grammar behaves weirdly in these cases. If you experiment a bit and look at the syntax tree (using the `:tree-sitter-subtree` and `:tree-sitter-scopes` commands), you might be able to find queries that cover most of the common situations though.

**Comment by majiayu000** [comment]
Hi, I've been looking into this and traced the root cause.

**Proposed approach:** 1. Modify runtime/queries/go/indents.scm to add error node matching: `(ERROR "{" @indent @extend)`. 2. Test with the reproduction steps from the issue (function body without closing brace, if statements). 3. Experiment with `:tree-sitter-subtree` to verify correct behavior for nested cases. 4. May also need `(ERROR "}" @outdent)` for dedent. Keep scope narrow — fix common single-level cases first, note limitations for deeply nested incomplete code in PR description.

I'll open a draft PR shortly. Happy to adjust based on your preference.

**Comment by majiayu000** [comment]
I've been looking into this issue. The root cause is that when `auto-pairs = false`, typing an opening delimiter like `{` without a matching `}` causes tree-sitter to produce an ERROR node instead of the expected AST node (like `block`). The Go indent queries don't match ERROR nodes, so indentation doesn't trigger.

My fix adds ERROR node matching to `runtime/queries/go/indents.scm` for `{`, `(`, and `[` — similar to how Swift, Koka, and Python indent queries already handle this. Will open a draft PR shortly.

**Comment by amandalewis1231-lgtm** [comment]
Thank you for the detailed update and for identifying the root cause
regarding the tree-sitter ERROR nodes when auto-pairs is disabled.

**Comment by lukehart182-gif** [comment]
Thank you for the update.
